src/proof/PGTransformationAlgorithms.cc

```diff
@@ -555,73 +555,64 @@ void ProofGraph::proofTransformAndRestructure(const double left_time, const int
     }
 
     double init_time = cpuTime();
-    assert(!(max_num_loops > 0 && left_time > 0));
+    assert(not (max_num_loops > 0 and left_time > 0));
     //Flag to check if in a loop at least a transformation has been applied
     bool some_transf_done;
     long curr_num_loops = 0;
-    RuleContext ra1, ra2;
     std::deque<clauseid_t> q;
-    clauseid_t id;
     // Main external loop
     do {
+        assert(isResetVisited1() and isResetVisited2());
         // Enqueue leaves first
         q.assign(leaves_ids.begin(), leaves_ids.end());
         some_transf_done = false;
         do {
```

Comment:
Can we change this to a `while (not q.empty())`?

Reply:
Yes! Changed.